torch/testing/_internal/common_methods_invocations.py

```diff
@@ -12090,6 +12090,11 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                    toleranceOverride({torch.chalf: tol(atol=6e-2, rtol=5e-2)}),
                    'TestCommon', 'test_complex_half_reference_testing',
                ),
+               DecorateInfo(
+                   toleranceOverride({torch.float32: tol(atol=1e-4, rtol=1e-5)}),
+                   'TestCompositeCompliance', 'test_forward_ad',
+                   active_if=IS_ARM64
```

Comment:
That is interesting. Is the non-arm version close to this tolerance value? 
Could there be a problem with our arm conv backend?

Reply:
Is that something we should dive into more or not then?

Reply:
I looked a bit more into this - it seems to only happen because we are calling into the `Winograd3x3Depthwise` conv backend, which is only enabled when ARM NEON exists.

Reply:
Ho we did have issue with this one being wrong before...
So only a very specific sample need this tolerance override?

Reply:
> So only a very specific sample need this tolerance override?

Yeah, it should only happen for the depthwise convolution case

Reply:
I added a comment on these findings but adding a sample-specific tolerance override doesn't seem very easy to do

Reply:
I'm a bit un-easy with just raising the tolerance here as we might be hiding a bug.
Could it be that forward AD calls this function with a speicific configuration that is not tested otherwise and it is wrong?

Reply:
Investigated some more - `Winograd3X3Depthwise`'s inaccuracy is still the root cause, but the reason why the issue surfaces for this particular test is that the path calculating the expected results run with tensors that have requires_grad=True, whereas the the path computing the actual path have inputs that are requires_grad=False. This results in them running with different backends as another requirement for Winograd to be enabled is that none of the inputs can require grad. Detaching before running forward resolves this issue.